trieste/acquisition/rule.py

```diff
@@ -1092,9 +1262,32 @@ def state_func(
                 acquisition_space = state.acquisition_space
 
             state_ = BatchTrustRegion.State(acquisition_space)
-            points = self._rule.acquire(acquisition_space, models, datasets=datasets)
 
-            return state_, points
+            # If the base rule is a sequence, run it sequentially for each subspace.
+            # See earlier comments.
+            if self._rules is not None:
+                _points = []
+                for subspace, rule in zip(subspaces, self._rules):
+                    _models = subspace.select_models(models)
+                    _datasets = subspace.select_datasets(datasets)
+                    assert _models is not None
+                    # Remap all local tags to global ones. One reason is that single model
+                    # acquisition builders expect OBJECTIVE to exist.
+                    _models = {
+                        LocalizedTag.from_tag(tag).global_tag: model
+                        for tag, model in _models.items()
+                    }
+                    if _datasets is not None:
+                        _datasets = {
+                            LocalizedTag.from_tag(tag).global_tag: dataset
+                            for tag, dataset in _datasets.items()
+                        }
+                    _points.append(rule.acquire(subspace, _models, _datasets))
+                points = tf.stack(_points, axis=1)
+            else:
+                points = self._rule.acquire(acquisition_space, models, datasets)
+
+            return state_, tf.reshape(points, [-1, len(subspaces), points.shape[-1]])
```

Comment:
So are we always returning rank 3 now, even if there are no subspaces? (or am I misunderstanding?)

Reply:
Yes that is correct. Which does mean that the caller has to use the new batched observer with this. `BayesianOptimizer`` class has rank detection in it now. For ask-tell optimiser, the user has to use the correct observer, e.g. see `integration/test_ask_tell_optimization.py` for similar rank detection.

Reply:
Is this a breaking change then? Our next release is probably going to include some anyway, so not necessarily an issue if it is, though we'll want to make sure we document it properly in the release notes.

Reply:
Updated the PR description.